### ISSUE TITLE
Optionally filter edit_post_link on frontend to deep-link to post in Customizer

### ISSUE THREAD

**Issue by westonruter** [xwp/wp-customize-posts#34]

**Comment by westonruter** [comment]
The `url` param should be added here as well and it should be the current URL on which the `edit_post_link()` appears, as opposed to the permalink for the post. This would allow the Customizer preview to initially load in the same view and when the link was clicked.

**Comment by stuartshields** [comment]
@westonruter This doesn't appear to support custom post types... Is that correct?

Even though the URL is formatted correctly, it gives us a error 404.
http://vip.local/wp-admin/customize.php?url=http%3A%2F%2Fvip.local%2F%3Fpreview%3Dtrue%26p%3D22&amp;previewed_post=22&amp;autofocus[section]=post[movies][22]&amp;customize_preview_post_nonce=8745488e02

**Comment by westonruter** [comment]
@stuartshields it does work with custom post types.

Your URL here looks like it has HTML encoding of entities.

**Comment by westonruter** [comment]
Good on you for ensuring the url of the post being previewed is also included in the edit post link, _with_ the autofocus to open the section when the customizer loads. I didn't note that in the description.

**Comment by stuartshields** [comment]
@westonruter I haven't done anything that was already in the code as references here: https://github.com/xwp/wp-customize-posts/blob/develop/php/class-edit-post-preview.php#L108

I don't like repeating myself so I'm thinking I'll build on what's already done. You can't see any issues with me doing this can you? I should have this and #93 ready by the end of the week. I also noticed when using this deep link with #93 that no Save/Publish button appears have you come across that issue?

**Comment by westonruter** [comment]
@stuartshields yes, extracting that logic into a separate `get_customize_url` method is great.

>  I also noticed when using this deep link with #93 that no Save/Publish button appears have you come across that issue?

Yes. I see that too. This is due to `\Edit_Post_Preview::can_load_customize_post_preview()` incorrectly returning `true` in this instance. Basically, since you're re-using the same logic to generate the URL to access that post in the preview, the existing URL generated is expecting that you are loading it _from_ the edit post screen via clicking the Preview Changes button. When opening the Customizer via the Preview Changes button, it disables the Save & Publish button and removes all controls/sections/panels not relevant to the current post. This is undesirable when clicking Edit in Customizer generally. The quickest workaround is to prevent `previewed_post` from being added as a query var expect in the URL generated for clicking the Preview Changes button.
